Workiva/ixbrl-viewer
# XT-2790 Fix escaping of style element contents (add custom XHTML serializer)

#### Reason for change

At present, the XHTML produced by the viewer builder escapes `<`, `&` and `>` throughout the document.  This is fine as long as the result is treated as XHTML.   Ordinary HTML expects different escaping to `<style>` and `<script>` elements.  In particular, an escaped `>` in a style element will not be recognised as valid CSS, resulting in different rendering in HTML and XHTML rendering mode.

#### Description of change

With this PR, the viewer does not escape `>` appearing within `<style>` elements (this is optional in XML/XHTML).

`<` and `&` are escaped as if they are appearing within a CSS string using `\NNNNNN` hex escapes (this is the only place other than comments in which they can legally appear in CSS).

This is an alternative solution for #384 .

#### Steps to Test

Create an iXBRL document with an inline style sheet, and a style rule such as:

```css
p > span { color: #f00 }
```

Create a viewer from it, and confirm that the `>` is not escaped in the resulting output.

**review**:
@Workiva/xt
@paulwarren-wk
